### PR TITLE
Fixing bug parsing ReportsEntryPoint CLI

### LINKED ISSUE #1816: `report` entry point input arguments are broken
The `report` entry point is supposed to take specific `(cycle, node)` combinations as an input argument, or a `max_node` or `min_node`. However, all of these inputs are parsed as strings, and are then later an attempt is made to compare `(cycle, node)` as a `tuple` to these strings. The check of the form `(int, int) not in str` fails because the two types cannot be compared.

https://github.com/terrapower/armi/blob/d7028672d345aadada35815ecbce8f2fdb0761aa/armi/cli/reportsEntryPoint.py#L120-L134

This code is not unit-tested at all, and it has presumably been broken for a long time.

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

When users use the `ReportsEntryPoint` CLI, they can pass in various integers. The CLI parser here never converts the CLI strings to integers, leading to an error if you use them.

## Why is the change being made?

To close #1816

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/tests/test_reactors.py [hunk 1]
```diff
@@ -834,8 +834,6 @@ def test_getDominantMaterial(self):
         self.assertEqual(dominantFuel.getName(), "UZr")
         self.assertEqual(dominantCool.getName(), "Sodium")
 
-        self.assertEqual(list(dominantCool.getNuclides()), ["NA23"])
```
Comment (mgjarrett) [inline armi/reactor/tests/test_reactors.py:837]: Why was this line deleted? Is it related to the main purpose of this PR, or is it in here by accident?

File: armi/utils/mathematics.py [hunk 2]
```diff
@@ -57,7 +64,7 @@ def average1DWithinTolerance(vals, tolerance=0.2):
     return avg
 
 
-def convertToSlice(x, increment=False):
+def convertToSlice(x, increment=0):
```
Comment (mgjarrett) [inline armi/utils/mathematics.py:67]: With this change, you could consider removing lines 121-122 below:

```
if increment is False:
    increment = 0
```

This would effectively change the API so we might want to avoid that, but I don't see any usage in our codebase where `increment=False` is passed explicitly.

**Comment by mgjarrett** [comment]
The bug fix and unit tests for the `report` entry point looks good to me. Some of the changes appear unrelated to this bug fix, at least on the surface. The specific files that seem possibly unrelated are:

- test_reactors.py
- mathematics.py
- doc/conf.py

Were these changes intended to be included? And if so, how do they relate to the reports entry point?

File: armi/utils/mathematics.py (hunk 2 above)
Comment (john-science) [reply armi/utils/mathematics.py:67]: Yeah, I was just trying to preserve the old API.

But... I have no idea _why_ anyone would want to pass in `False` here. This is a math function, after all.

Hmmm

**Comment by john-science** [comment]
> Were these changes intended to be included? And if so, how do they relate to the reports entry point?

Whoops! The last commit was to the wrong branch. One sec, I'll revert that.

**Review by mgjarrett — APPROVED** [review]

File: armi/reactor/tests/test_reactors.py (hunk 1 above)
Comment (john-science) [reply armi/reactor/tests/test_reactors.py]: That is part of another PR now, I removed that line from this one.

Ignore it.

(It was because of a `DeprecationWarning`.)

File: armi/utils/mathematics.py (hunk 2 above)
Comment (john-science) [reply armi/utils/mathematics.py:60]: Okay, I removed that strange `False` from the API, and I am running a big suite of downstream unit tests.

Comment (john-science) [reply armi/utils/mathematics.py:60]: Success!  Okay, done.

**Review by mgjarrett — APPROVED** [review]
